ci/run-unit-tests.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -o errexit -o pipefail -o nounset
+
+if [[ ! -d CMakeFiles ]]; then
+    echo "Warning: This script should be run from within a build directory" 1>&2
+fi
+
+if [[ $# -ne 0 ]]; then
+    echo "Usage: $0" >&2
+    exit 1
+fi
+
+# Running "make test" is slow (why?), so we just call all test executables manually
```

Comment:
I guess you could remove the `(why?)` part, it's because CTest will run each test case in a separate process, incurring the non-negligible start up costs each time.

Reply:
I've added the explanation to the comment.